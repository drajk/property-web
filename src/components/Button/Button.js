import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './Button.module.css';

const Button = ({ children, isActive, onClick }) => {
    const cxContent = cx(styles.content, {
        [styles.isActive]: isActive,
    });

    return (
        <div className={styles.container} onClick={onClick}>
            <span className={cxContent}>{children}</span>
        </div>
    );
};

Button.propTypes = {
    children: PropTypes.node.isRequired,
    isActive: PropTypes.bool,
    onClick: PropTypes.func,
};

Button.defaultProps = {
    isActive: false,
    onClick: () => {},
};

export default Button;
