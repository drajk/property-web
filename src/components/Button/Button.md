# Button.js

Shows a styled link button.

## Usage

```html
<Button>Search</Button>
<Button isActive>Search</Button>
<Button isActive onClick={() => {}}>Search</Button>
```
