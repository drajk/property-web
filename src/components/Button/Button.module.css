.container {
    display: flex;
    align-items: center;
    cursor: pointer;
}

.content {
    border-radius: 4px;
    font-size: var(--font-size-small);
    width: auto;
    font-family: var(--font-family);
    padding: 12px 20px;
}

.content:hover {
    color: var(--active-color);
}

.isActive {
    background-color: var(--border-color);
    font-weight: 700;
    color: var(--active-color);
}