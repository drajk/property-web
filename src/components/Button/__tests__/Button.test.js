import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button', () => {
    it('renders with text', () => {
        const tree = renderer.create(<Button>Search</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with active state', () => {
        const tree = renderer.create(<Button isActive>Search</Button>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with active state and handler', () => {
        const tree = renderer
            .create(
                <Button isActive onClick={() => {}}>
                    Search
                </Button>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
