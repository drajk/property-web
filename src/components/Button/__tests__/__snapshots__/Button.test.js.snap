// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders with active state 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <span
    className="content isActive"
  >
    Search
  </span>
</div>
`;

exports[`Button renders with active state and handler 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <span
    className="content isActive"
  >
    Search
  </span>
</div>
`;

exports[`Button renders with text 1`] = `
<div
  className="container"
  onClick={[Function]}
>
  <span
    className="content"
  >
    Search
  </span>
</div>
`;
